save-orchestrator/src/main/kotlin/org/cqfn/save/orchestrator/service/DockerService.kt

```diff
@@ -240,6 +254,63 @@ class DockerService(private val configProperties: ConfigProperties) {
         return Triple(imageId, agentRunCmd, saveCliExecFlags)
     }
 
+    @Suppress("TOO_MANY_LINES_IN_LAMBDA")
+    private fun unzipArchivesAmongAdditionalFiles(
+        execution: Execution,
+        isStandardMode: Boolean,
+        testSuitesDir: File,
+        resourcesPath: File) {
+        // FixMe: for now support only .zip files
+        execution.additionalFiles?.split(";")?.filter { it.endsWith(".zip") }?.forEach {
+            val fileLocation = if (isStandardMode) {
+                testSuitesDir
+            } else {
+                val testRootPath = webClientBackend.post()
+                    .uri("/findTestRootPathForExecutionByTestSuites")
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .body(BodyInserters.fromValue(execution))
+                    .retrieve()
+                    .bodyToMono<List<String>>()
+                    .block()!!
+                    .distinct()
+                    .single()
+                resourcesPath.resolve(testRootPath)
+            }
+
+            val file = fileLocation.resolve(File(it).name)
+
+            val isValidZipArchive = file.isValidZipArchive()
+
+            if (isValidZipArchive) {
+                log.debug("Unzip ${file.absolutePath} into ${fileLocation.absolutePath}")
+                file.unzipInto(fileLocation)
+                file.delete()
+            }
+        }
+    }
+
+    @Suppress("MAGIC_NUMBER", "LONG_NUMERICAL_VALUES_SEPARATED", "MagicNumber")
+    private fun File.isValidZipArchive(): Boolean {
+        var fileSignature: Int
+        try {
+            RandomAccessFile(this, "r").use { raf -> fileSignature = raf.readInt() }
+        } catch (e: IOException) {
+            log.error("Error during checking file ${this.name} for being an archive: ${e.message}")
+            fileSignature = 0
+        }
+        // check for 'magic numbers', i.e. the data used to identify or verify the content of a file
+        return fileSignature == 0x504B0304 || fileSignature == 0x504B0506 || fileSignature == 0x504B0708
```

Comment:
Doesn't the library handle such checks by itself? Why is this check needed?

Reply:
Yeah, let it be covered by library